hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestBlockDataStreamOutput.java

```diff
@@ -105,90 +132,109 @@ public static void init() throws Exception {
+        Arguments.of(false, false)
+    );
   }
 
-  @AfterAll
-  public static void shutdown() {
-    IOUtils.closeQuietly(client);
-    if (cluster != null) {
-      cluster.shutdown();
-    }
+  private static Stream<Arguments> dataLengthParameters() {
+    return Stream.of(
+        Arguments.of(chunkSize / 2),
+        Arguments.of(chunkSize),
+        Arguments.of(chunkSize + 50),
+        Arguments.of(blockSize + 50)
+    );
   }
 
-  @Test
-  public void testHalfChunkWrite() throws Exception {
-    testWrite(chunkSize / 2);
-    testWriteWithFailure(chunkSize / 2);
+  static OzoneClientConfig newClientConfig(ConfigurationSource source,
+                                           boolean flushDelay, boolean enablePiggybacking) {
+    OzoneClientConfig clientConfig = source.getObject(OzoneClientConfig.class);
+    clientConfig.setChecksumType(ContainerProtos.ChecksumType.NONE);
+    clientConfig.setStreamBufferFlushDelay(flushDelay);
+    clientConfig.setEnablePutblockPiggybacking(enablePiggybacking);
+    return clientConfig;
   }
 
-  @Test
-  public void testSingleChunkWrite() throws Exception {
-    testWrite(chunkSize);
-    testWriteWithFailure(chunkSize);
+  static OzoneClient newClient(OzoneConfiguration conf,
+                               OzoneClientConfig config) throws IOException {
+    OzoneConfiguration copy = new OzoneConfiguration(conf);
+    copy.setFromObject(config);
+    return OzoneClientFactory.getRpcClient(copy);
+  }
+
+  /**
+   * Create a MiniDFSCluster for testing.
+   * <p>
+   * Ozone is made active by setting OZONE_ENABLED = true
+   *
+   * @throws IOException
+   */
+  @BeforeAll
+  public void init() throws Exception {
+    cluster = createCluster();
   }
 
-  @Test
-  public void testMultiChunkWrite() throws Exception {
-    testWrite(chunkSize + 50);
-    testWriteWithFailure(chunkSize + 50);
+  static String getKeyName() {
+    return UUID.randomUUID().toString();
   }
 
-  @Test
-  @Flaky("HDDS-12027")
-  public void testMultiBlockWrite() throws Exception {
-    testWrite(blockSize + 50);
-    testWriteWithFailure(blockSize + 50);
+  /**
+   * Shutdown MiniDFSCluster.
+   */
+  @AfterAll
+  public void shutdown() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
   }
 
-  static void testWrite(int dataLength) throws Exception {
-    XceiverClientMetrics metrics =
-        XceiverClientManager.getXceiverClientMetrics();
-    long pendingWriteChunkCount = metrics.getPendingContainerOpCountMetrics(WriteChunk);
-    long pendingPutBlockCount = metrics.getPendingContainerOpCountMetrics(PutBlock);
+  @ParameterizedTest
+  @MethodSource("dataLengthParameters")
+  public void testHalfChunkWrite(int dataLength) throws Exception {
+    OzoneClientConfig config = newClientConfig(cluster.getConf(), false, true);
+    try (OzoneClient client = newClient(cluster.getConf(), config)) {
+      testWrite(client, dataLength);
+      testWriteWithFailure(client, dataLength);
+    }
+  }
```

Comment:
There is discrepancy here. The method name is `testHalfChunkWrite`, but `dataLengthParameters` here have different lengths.

Can use a more generalized name like `testStreamWrite`.

In the future, we can separate the different data length to different methods since each cases might have specialized assertions only for that case (e.g. in `BlockOutputStream`: `testWriteMoreThanChunkSize`, `testWriteExactlyFlushSize`, etc.)